reactor-netty-core/src/main/java/reactor/netty/channel/ChannelOperations.java

```diff
@@ -453,7 +451,7 @@ protected final void terminate() {
 			// when there is no response state
 			onInboundComplete();
 			afterInboundComplete();
-			onTerminate.onComplete();
+			onTerminate.emitEmpty(Sinks.EmitFailureHandler.FAIL_FAST);
```

Comment:
I think `tryEmit()` could be sufficient here. If there are competing attempts to terminate, I'm guessing it doesn't matter which one wins? If that is the case you can also initialize via `Sinks.unsafe().empty()` and avoid unnecessary protection.

Reply:
It is guaranteed that will be invoked in an event loop, so I think `Sinks.unsafe().empty()` should be OK.
Your suggestion is to change from `emitEmpty(Sinks.EmitFailureHandler.FAIL_FAST)` to `tryEmit`?

Reply:
please check